### ISSUE TITLE
Schema's are getting deleted

### ISSUE THREAD

**Issue by BharathMannem** [Azure/azure-api-management-devops-resource-kit#220]
Hi Team, 

ExtractTool is generating the schema definitions present in the apim, but when deployed it is deleting the existing one.

**Comment by miaojiang** [comment]
Hi @BharathMannem  can you please elaborate? 

If the schema is present in the ARM templates, it should not delete the existing one.

**Comment by patros** [comment]
I've hit this one too. The schema is only partially present.

Example...
```
{
    "type": "Microsoft.ApiManagement/service/apis/schemas",
    "apiVersion": "2019-01-01",
    "name": "[concat(parameters('ApiManagementName'), '/sms/5d671ee8543bd40250047d8e')]",
    "dependsOn": [
        "[resourceId('Microsoft.ApiManagement/service/apis', parameters('ApiManagementName'), 'sms')]"
    ],
    "properties": {
        "contentType": "application/vnd.oai.openapi.components+json",
        "document": {}
    }
}
```

Notice the actual json schema content is missing.

I believe the resource is still present as a result. It just isn't visible in the portal.

If you export the resource group template via the portal or the az cli the json schema content is still missing. So not specifically a problem with the tool here. But one with the RM API I suspect.

**Comment by patros** [comment]
Hmm... I'm getting different behaviour from different instances. One will export the document the other doesn't.

**Comment by BharathMannem** [comment]
@miaojiang I tried the same scenario more than 10times the ARM template generated is deleting the existing definitions. 
@patros  I am getting the right content in the generated arm template but it is deleting the existing definitions.

**Comment by bremnes** [comment]
Perhaps you guys can take a look at what is actually stored in Azure? The extraction tool calls these API's and extracts schema(s) based on what's in Azure.

If you go to https://resources.azure.com/raw you can do ARM debugging there. If you try with the following query (replace values within [] with your own), then you should be able to see the raw data stored in Azure. We've used this do figure out some issues.

>GET https://management.azure.com/subscriptions/[SUBSCRIPTION-ID]/resourceGroups/[RESOURCEGROUPNAME]/providers/Microsoft.ApiManagement/service/[API-MANAGEMENT-INSTANCE-NAME]/apis/[API-ID]/schemas?api-version=2019-01-01

(Mark the query above and click CTRL+S to send it, then you should get the result back in the right pane.)

**Comment by bremnes** [comment]
@BharathMannem, @coderpatros, could you try to re-create this using the latest version? We've fixed some other schema issues, so it would be interesting to see if this fixes your issue.

If you are still experiencing the error, are you able to
* Show us the relevant sections of the extracted ARM templates? Mainly the resource with type `Microsoft.ApiManagement/service/apis/schemas` _before_ you've done any extraction so that we can take a look at the original.

Are all API's you're experiencing this for of type OpenAPI v3? We previously had some issues extracting swagger and wsdl api's, and this might be similar for OpenAPI.

**Comment by UmairSyed** [comment]
@BharathMannem @coderpatros I have found a similar issue when the type is "application/vnd.ms-azure-apim.swagger.definitions+json". basically the extractor is producing the following json format:
        "document": {
          "value": "{\"TokenGenerateResponse\":{\"type\":\..."
        }
however the ARM api expects:
        "document": {
          "definitions": "{\"TokenGenerateResponse\":{\"type\":\..."
        }
I think this behaviour is also discussed here:
https://github.com/Azure/azure-rest-api-specs/issues/6372
I have updated the extractor to cater this scenario already in my fork so if there are no concerns I can create a PR.

**Comment by bremnes** [comment]
@BharathMannem, there has been some code changes which has solved similar problems earlier, see #264 and #190. They match your description with schemas being gone.

I'm closing this issue as I _think_ it's solved, but feel free to re-open if you are still having problems.
